include/swift/AST/ModuleDependencies.h

```diff
@@ -935,17 +935,26 @@ using ModuleDependenciesKindRefMap =
 class SwiftDependencyTracker {
 public:
   SwiftDependencyTracker(llvm::cas::CachingOnDiskFileSystem &FS,
-                         llvm::TreePathPrefixMapper *Mapper)
-      : FS(FS.createProxyFS()), Mapper(Mapper) {}
+                         llvm::TreePathPrefixMapper *Mapper,
+                         const CompilerInvocation &CI);
```

Comment:
Throughout this PR I see you are still using `TreePathPrefixMapper`, but that requires a casfs. Should this change to a normal `PrefixMapper`?

Reply:
Scanner is still using a CASFS to scan but builds into a IncludeTreeFileList. I can't just switch to a `CASProvidingFileSystem` because that has to use together with a FileSystemStatsCache, otherwise it is going to be super expensive.

Let me see if I can switch to the simpler prefix mapper and if there are any side-effects.